MagneticField/GeomBuilder/src/bRod.h

```diff
@@ -8,30 +8,32 @@
  *  \author N. Amapane - INFN Torino
  */
 
-#include "MagneticField/GeomBuilder/src/MagGeoBuilderFromDDD.h"
-#include "MagneticField/GeomBuilder/src/volumeHandle.h"
-#include "MagneticField/GeomBuilder/src/bSlab.h"
+#include "bSlab.h"
 
 class MagBRod;
 
-class MagGeoBuilderFromDDD::bRod {
-public:
-  /// Constructor from list of volumes
-  bRod(handles::const_iterator begin, handles::const_iterator end);
+namespace magneticfield {
 
-  /// Destructor
-  ~bRod();
+  class bRod {
+  public:
+    /// Constructor from list of volumes
+    bRod(handles::const_iterator begin, handles::const_iterator end, bool debugVal = false);
 
-  /// Distance from center along sector normal.
-  const float RN() const { return volumes.front()->RN(); }
+    /// Destructor
+    ~bRod() = default;
 
-  /// Construct the MagBRod upon request.
-  MagBRod* buildMagBRod() const;
+    /// Distance from center along sector normal.
+    const float RN() const { return volumes.front()->RN(); }
 
-private:
-  std::vector<bSlab> slabs;
-  handles volumes;  // pointers to all volumes in the rod
-  mutable MagBRod* mrod;
-};
+    /// Construct the MagBRod upon request.
+    MagBRod* buildMagBRod() const;
+
+  private:
+    std::vector<bSlab> slabs;
+    handles volumes;  // pointers to all volumes in the rod
+    mutable MagBRod* mrod;
+    bool debug;  // Allow assignment from other bRod objects
```

Comment:
and here is a result of that static member change. The size of bool is 1 byte, but how many bRod and other objects are there?

Reply:
The most ubiquitous are volumeHandle which are of the order of ~2k; the total number of concerned instances is probably less than 3k. Consider that these are temporary objects, and are deleted after the map construction is done.